src/boosting/gbdt.cpp

```diff
@@ -419,20 +419,15 @@ bool GBDT::TrainOneIter(const score_t* gradients, const score_t* hessians) {
     } else {
       // only add default score one-time
       if (models_.size() < static_cast<size_t>(num_tree_per_iteration_)) {
-        double output = 0.0;
-        if (!class_need_train_[cur_tree_id]) {
```

Comment:
Note that in `GBDT::BoostFromAverage`, the average score is calculated and added to the training/validation scorers regardless of `class_need_train_`, so this should be remove.